vllm/model_executor/layers/sampler.py

```diff
@@ -120,7 +123,7 @@ def _apply_min_tokens_penalty(
         if (i < sampling_metadata.num_prompts
                 and sampling_params.prompt_logprobs is not None):
             assert len(seq_ids) == 1
-            start_idx += sampling_metadata.prompt_lens[i] - 1
+            start_idx += sampling_metadata.subquery_lens[i] - 1
```

Comment:
we should use subquery len here. Prompt_len contains entire prompt token's length